src/parser_urdf.cc

```diff
@@ -3267,44 +3299,41 @@ TiXmlDocument URDF2SDF::InitModelString(const std::string &_urdfStr,
     }
     catch(...)
     {
-      delete sdf;
       throw;
     }
   }
   catch(...)
   {
-    delete robot;
     throw;
   }
 
-  sdfXmlOut.LinkEndChild(sdf);
-
-  return sdfXmlOut;
+  _sdfXmlOut->LinkEndChild(sdf);
 }
 
 ////////////////////////////////////////////////////////////////////////////////
-TiXmlDocument URDF2SDF::InitModelDoc(TiXmlDocument* _xmlDoc)
+void URDF2SDF::InitModelDoc(const tinyxml2::XMLDocument *_xmlDoc,
+                            tinyxml2::XMLDocument *_sdfXmlDoc)
 {
-  std::ostringstream stream;
-  stream << *_xmlDoc;
-  std::string urdfStr = stream.str();
-  return InitModelString(urdfStr);
+  tinyxml2::XMLPrinter printer;
+  _xmlDoc->Print(&printer);
+  std::string urdfStr = printer.CStr();
+  InitModelString(urdfStr, _sdfXmlDoc);
 }
 
 ////////////////////////////////////////////////////////////////////////////////
-TiXmlDocument URDF2SDF::InitModelFile(const std::string &_filename)
+void URDF2SDF::InitModelFile(const std::string &_filename,
+                             tinyxml2::XMLDocument *_sdfXmlDoc)
 {
-  TiXmlDocument xmlDoc;
-  if (xmlDoc.LoadFile(_filename))
+  tinyxml2::XMLDocument xmlDoc;
+  if (!xmlDoc.LoadFile(_filename.c_str()))
   {
-    return this->InitModelDoc(&xmlDoc);
+    return this->InitModelDoc(&xmlDoc, _sdfXmlDoc);
```

Comment:
this is a `void` function; perhaps move `this->InitModelDoc(&xmlDoc, _sdfXmlDoc);` to the previous line?

Reply:
I'll just drop the return from here, it's unnecessary.